pkg/stanza/fileconsumer/file.go

```diff
@@ -261,3 +270,14 @@ func (m *Manager) newReader(ctx context.Context, file *os.File, fp *fingerprint.
 	m.telemetryBuilder.FileconsumerOpenFiles.Add(ctx, 1)
 	return r, nil
 }
+
+func (m *Manager) instantiateTracker(persister operator.Persister) {
+	opts := []tracker.OptionFunc{tracker.WithMaxBatchFiles(m.maxBatchFiles)}
+	if m.noTracking {
+		opts = append(opts, tracker.WithNoTracking())
+	}
+	if m.pollsToArchive > 0 {
+		opts = append(opts, tracker.WithPollsToArchive(m.pollsToArchive), tracker.WithPersister(persister))
+	}
+	m.tracker = tracker.NewFileTracker(m.set, opts...)
```

Comment:
It seems unnecessary to change the pattern we already had. If there's no tracking, just instantiate a `NewNoStateTracker`.

If there is tracking, just pass it the batch size and persister (even if it is nil). I don't think there's enough complexity to justify the tracker having its own options yet.

Inside the tracker, we can make decisions about archiving based on whether the persister is nil.

Reply:
Yeah, that works. It's probably the best option considering and we can change the pattern in future, if needed.

Reply:
Done. Looks much simple. Thanks!